File: fix-all-conflicts-comprehensive.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { execSync } from 'child_process';
console.log('🔧 Starting comprehensive merge conflict resolution...');

// Function to fix merge conflicts in a file
function fixMergeConflicts(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Skip if no merge conflicts
    if (!content.includes(')
    content = content.replace(/<<<<<<< HEAD[\s\S]*?=======([\s\S]*?)
    
    // Remove any remaining conflict markers
    content = content.replace(/[\s\S]*?
    
    // Clean up extra newlines
    content = content.replace(/\n\s*\n\s*\n/g, '\n\n');
    
    fs.writeFileSync(filePath, content);
    return true;
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
    return false;
  }
}

// Function to recursively find all files with merge conflicts
function findFilesWithConflicts(dir) {
  const files = [];
  
  function traverse(currentDir) {
    const items = fs.readdirSync(currentDir);
    
    for (const item of items) {
      const fullPath = path.join(currentDir, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
        traverse(fullPath);
      } else if (stat.isFile() && (item.endsWith('.tsx') || item.endsWith('.ts') || item.endsWith('.js') || item.endsWith('.jsx'))) {
        try {
          const content = fs.readFileSync(fullPath, 'utf8');
          if (content.includes('<<<<<<< HEAD')) {
            files.push(fullPath);
          }
        } catch (error) {
          // Skip files that can't be read
        }
      }
    }
  }
  
  traverse(dir);
  return files;
}

// Find all files with conflicts
const conflictedFiles = findFilesWithConflicts('./');
console.log(`Found ${conflictedFiles.length} files with merge conflicts`);

// Fix each file
let fixedCount = 0;
for (const file of conflictedFiles) {
  if (fixMergeConflicts(file)) {
    fixedCount++;
  }
}

console.log(`✅ Fixed merge conflicts in ${fixedCount} files`);

// Additional cleanup for common issues
console.log('🧹 Performing additional cleanup...');

// Fix common import issues
const commonFixes = [
  {
    pattern: /import\s+{\s*}\s+from\s+['"][^'"]+['"];?\s*\n/g,
    replacement: ''
  },
  {
    pattern: /import\s+[^;]+;\s*\n\s*import\s+[^;]+;\s*\n\s*import\s+[^;]+;\s*\n/g,
    replacement: (match) => {
      const imports = match.split('\n').filter(line => line.trim());
      const uniqueImports = [...new Set(imports)];
      return uniqueImports.join('\n') + '\n';
    }
  },
  {
    pattern: /\n\s*\n\s*\n+/g,
    replacement: '\n\n'
  }
];

// Apply common fixes to all TypeScript/JavaScript files
function applyCommonFixes(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    for (const fix of commonFixes) {
      const newContent = content.replace(fix.pattern, fix.replacement);
      if (newContent !== content) {
        content = newContent;
        modified = true;
      }
    }
    
    if (modified) {
      fs.writeFileSync(filePath, content);
      return true;
    }
  } catch (error) {
    // Skip files that can't be processed
  }
  return false;
}

// Apply fixes to all relevant files
const allFiles = findFilesWithConflicts('./');
let cleanedCount = 0;

for (const file of allFiles) {
  if (applyCommonFixes(file)) {
    cleanedCount++;
  }
}

console.log(`🧹 Applied additional cleanup to ${cleanedCount} files`);

// Verify no more conflicts
const remainingConflicts = findFilesWithConflicts('./');
if (remainingConflicts.length === 0) {
  console.log('✅ All merge conflicts resolved successfully!');
} else {
  console.log(`⚠️  ${remainingConflicts.length} files still have conflicts:`);
  remainingConflicts.forEach(file => console.log(`  - ${file}`));
}

console.log('🎉 Merge conflict resolution complete!');

File: fix_all_conflicts_comprehensive.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_merge_conflicts(file_path):
    """Fix merge conflicts in a single file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        if '<<<<<<< HEAD' not in content:
            return False
            
        # Remove all merge conflict markers and keep the HEAD version
        # Pattern 1: <<<<<<< HEAD ... ======= ... >>>>>>> branch
        content = re.sub(r'<<<<<<< HEAD\n(.*?)\n=======\n(.*?)\n>>>>>>> [^\n]+\n?', r'\1', content, flags=re.DOTALL)
        
        # Pattern 2: <<<<<<< HEAD ... ======= ... >>>>>>> branch (without newlines)
        content = re.sub(r'<<<<<<< HEAD\n(.*?)\n=======\n(.*?)\n>>>>>>> [^\n]+', r'\1', content, flags=re.DOTALL)
        
        # Remove any remaining conflict markers
        content = re.sub(r'<<<<<<< HEAD\n?', '', content)
        content = re.sub(r'=======\n?', '', content)
        content = re.sub(r'>>>>>>> [^\n]+\n?', '', content)
        
        # Clean up any double newlines
        content = re.sub(r'\n\n\n+', '\n\n', content)
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        
        return True
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Fix merge conflicts in all relevant files"""
    patterns = [
        '**/*.tsx',
        '**/*.ts', 
        '**/*.js',
        '**/*.jsx'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            # Skip node_modules and other irrelevant directories
            if any(skip in file_path for skip in ['node_modules', '.git', 'dist', 'build', '.next']):
                continue
                
            files_processed += 1
            if fix_merge_conflicts(file_path):
                files_fixed += 1
                print(f"Fixed conflicts in: {file_path}")
    
    print(f"\nProcessed {files_processed} files, fixed conflicts in {files_fixed} files")

if __name__ == "__main__":
    main()